qa/integration-tests/src/test/java/io/camunda/it/client/ProcessQueryTest.java

```diff
@@ -0,0 +1,567 @@
+
+    // then
+    assertThat(result.items().size()).isEqualTo(expectedBpmnProcessIds.size());
+    assertThat(result.items().stream().map(ProcessInstance::getBpmnProcessId).toList())
+        .containsExactlyInAnyOrderElementsOf(expectedBpmnProcessIds);
+  }
+
+  @Test
+  void shouldRetrieveProcessInstancesByBpmnProcessId() {
+    // given
+    final String bpmnProcessId = "demo_process_v1";
+    final long processInstanceKey =
+        PROCESS_INSTANCES.stream()
+            .filter(p -> Objects.equals(bpmnProcessId, p.getBpmnProcessId()))
+            .findFirst()
+            .orElseThrow()
+            .getProcessInstanceKey();
+
+    // when
+    final var result =
+        zeebeClient
+            .newProcessInstanceQuery()
+            .filter(f -> matchAllProcessInstancesFilter(f).bpmnProcessId(bpmnProcessId))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(result.items().get(0).getKey()).isEqualTo(processInstanceKey);
+  }
+
+  @Test
+  void shouldRetrieveProcessInstancesByStateRunning() {
+    // when
+    final var result =
+        zeebeClient
+            .newProcessInstanceQuery()
+            .filter(f -> f.running(true).active(true).incidents(true))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(3);
+    assertThat(result.items().stream().map(ProcessInstance::getBpmnProcessId).toList())
+        .containsExactlyInAnyOrder("demo_process_v1", "demo_process_v2", "incident_process_v1");
+  }
+
+  @Test
+  void shouldRetrieveProcessInstancesByVariable() {
+    // given
+    final ProcessInstanceVariableFilterRequest variableFilter =
+        new ProcessInstanceVariableFilterRequest().name("path").addValuesItem("222");
+
+    // when
+    final var result =
+        zeebeClient
+            .newProcessInstanceQuery()
+            .filter(f -> matchAllProcessInstancesFilter(f).variable(variableFilter))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(result.items().stream().map(ProcessInstance::getBpmnProcessId).toList())
+        .containsExactlyInAnyOrder("demo_process_v2");
+  }
+
+  @Test
+  void shouldRetrieveNoProcessInstanceOnVariableValueMismatch() {
+    // given
+    final ProcessInstanceVariableFilterRequest variableFilter =
+        new ProcessInstanceVariableFilterRequest().name("path").addValuesItem("123");
+
+    // when
+    final var result =
+        zeebeClient
+            .newProcessInstanceQuery()
+            .filter(f -> matchAllProcessInstancesFilter(f).variable(variableFilter))
+            .send()
+            .join();
+
+    // then
+    assertThat(result.items()).isEmpty();
+  }
+
+  @Test
+  void shouldRetrieveNoProcessInstancesWhenNoFilter() {
+    // when
+    final var result = zeebeClient.newProcessInstanceQuery().send().join();
+
+    // then
+    assertThat(result.items()).isEmpty();
```

Comment:
It is strange that a search with no filters returns empty result
But, if it is specified this way...

Reply:
This is the default implementation in Operate, so I did the same. FYI @ralfpuchert